simpleclient_hotspot/src/main/java/io/prometheus/client/hotspot/ThreadExports.java

```diff
@@ -73,8 +73,38 @@ void addThreadMetrics(List<MetricFamilySamples> sampleFamilies) {
         "jvm_threads_deadlocked_monitor",
         "Cycles of JVM-threads that are in deadlock waiting to acquire object monitors",
         nullSafeArrayLength(threadBean.findMonitorDeadlockedThreads())));
+
+    Map<Thread.State, Integer> threadStateCounts = getThreadStateCountMap();
+    for (Map.Entry<Thread.State, Integer> entry : threadStateCounts.entrySet()) {
+      String stateName = entry.getKey().toString().toLowerCase();
+      sampleFamilies.add(
+        new GaugeMetricFamily(
+          String.format("jvm_threads_%s", stateName),
+          String.format("Current count of threads in state '%s'", stateName),
+          entry.getValue().doubleValue()));
+    }
   }
 
+  private Map<Thread.State, Integer> getThreadStateCountMap() {
+    ThreadInfo[] allThreads = threadBean.getThreadInfo(threadBean.getAllThreadIds(), StackTraceDepth);
+    HashMap<Thread.State, Integer> threadCounts = new HashMap<Thread.State, Integer>();
```

Comment:
These timeseries will not always be present, you should create all known states in advance